modules/460-log-shipper/hooks/generate_config.go

```diff
@@ -212,6 +330,17 @@ func generateConfig(input *go_hook.HookInput) error {
 	for _, destination := range destSnap {
 		dest := destination.(v1alpha1.ClusterLogDestination)
 
+		// Override TLS config if secretRef is provided
+		if secretName := getTLSSecretRef(dest); secretName != "" {
+			tlsConfig, err := getTLSConfigFromSecretSnapshot(secretName, tlsSecretsSnap)
+			if err != nil {
+				return errors.Wrap(err, "failed to get TLS config from secret snapshot")
+			}
+			if err := applyTLSConfig(&dest, tlsConfig); err != nil {
+				return errors.Wrap(err, "failed to apply TLS config")
+			}
+		}
+
```

Comment:
Here we take a pointer to the destination's TLS config struct, check if a user wants to use secretRef, and update appropriate fields with the data from the snapshot.

```suggestion
		// Override TLS config if secretRef is provided
		destTLSConfig := getTLSConfig(dest)
		if destTLSConfig != nil {
			if destTLSConfig.SecretRef != nil && destTLSConfig.SecretRef.Name != "" {
				tlsConfig, err := getTLSConfigFromSecretSnapshot(destTLSConfig.SecretRef.Name, tlsSecretsSnap)
				if err != nil {
					return errors.Wrap(err, "failed to get TLS config from secret snapshot")
				}

				applyTLSConfig(destTLSConfig, tlsConfig)
```